driver/src/main/java/org/neo4j/driver/v1/GraphDatabase.java

```diff
@@ -215,25 +214,33 @@ private static SecurityPlan createSecurityPlan( BoltServerAddress address, Confi
             throws GeneralSecurityException, IOException
     {
         Config.EncryptionLevel encryptionLevel = config.encryptionLevel();
-        boolean requiresEncryption = encryptionLevel.equals( REQUIRED ) ||
-                                     (encryptionLevel.equals( REQUIRED_NON_LOCAL ) && !address.isLocal());
+        boolean requiresEncryption = encryptionLevel.equals( REQUIRED );
 
         if ( requiresEncryption )
         {
             Logger logger = config.logging().getLog( "session" );
             switch ( config.trustStrategy().strategy() )
             {
+
+            // DEPRECATED CASES //
+            case TRUST_ON_FIRST_USE:
+                logger.warn(
+                        "Option `TRUST_ON_FIRST_USE` has been deprecated and will be removed in a future " +
+                        "version of the driver. Please switch to use `TRUST_ALL_CERTIFICATES` instead." );
+                return SecurityPlan.forTrustOnFirstUse( config.trustStrategy().certFile(), address, logger );
             case TRUST_SIGNED_CERTIFICATES:
                 logger.warn(
                         "Option `TRUST_SIGNED_CERTIFICATE` has been deprecated and will be removed in a future " +
-                        "version " +
-                        "of the driver. Please switch to use `TRUST_CUSTOM_CA_SIGNED_CERTIFICATES` instead." );
-                //intentional fallthrough
+                        "version of the driver. Please switch to use `TRUST_CUSTOM_CA_SIGNED_CERTIFICATES` instead." );
+                // intentional fallthrough
+            // END OF DEPRECATED CASES //
+
+            case TRUST_ALL_CERTIFICATES:
```

Comment:
Shall we default to `TRUST_ALL` for old `TRUST_SIGNED`? isn't it more reasonable to fall through to `TRUST_CUSTOM_CA`? The `log.warn` at line 233 suggests it would be `TRUST_CUSTOM_CA`